Consumer/MessageConsumer.php

```diff
@@ -50,22 +52,45 @@ class MessageConsumer implements ConsumerInterface
     private $serializer;
 
     /**
+     * Producer used to resend message to the end of the queue
+     *
+     * @var Producer
+     */
+    private $resendProducer;
+
+    /**
+     * Producer used to send message to quarantine queue
+     *
+     * @var Producer
+     */
+    private $quarantineProducer;
+
+    /**
      * Constructor
      *
-     * @param MessageDispatcher   $messageDispatcher
-     * @param SerializerInterface $serializer
+     * @param MessageDispatcher   $messageDispatcher  Message dispatcher
+     * @param SerializerInterface $serializer         Message serializer
+     * @param Producer            $resendProducer     Producer used to resend message to the end of the queue
+     * @param Producer            $quarantineProducer Producer used to send message to quarantine queue
```

Comment:
In addition I'm not sure why we need two separate queues here, just use a dead letter for both cases.

The resend producer also means we will have to have producer config in cases where the producer is in one project and the consumer in another.